text/0000-svelte-config-util.md

```diff
@@ -0,0 +1,165 @@
+to tools reading svelte config without `@sveltejs/config`.
+To reduce friction, v1.0 of @sveltejs/config could be limited to features that are explicitly backwards compatible
+and features that would break will only be released after our tools adopted it
+
+#### New Features for tool authors
+
+##### async `loadConfig` function
+```js
+import { loadConfig } from '@sveltejs/config';
+//...
+const config = await load_config();
+```
+
+loadConfig accepts an optional `options` argument with the following options:
+```js
+options = {
+    fromDir: '/some/path', // process.cwd by default, tools can pass what they need, eg vite root
+    inlineConfig: {/*...*/}, // svelte config object built from inline options eg cli args passed to the tool. merged in
+    namespace: 'kit', // namespace for the tool loading the config, useful for advanced features
+    //...
+}
+```
+
+##### defined namespacing (top-level keys owned by a tool)
+
+Currently, namespaces in svelte.config.js are loosely agreed upon. 
+`compilerOptions`, `preprocess`, `extensions` and `kit` are used by SvelteKit.
+
+Each tool will get its own namespace, e.g. `vitePlugin`, `rollupPlugin`, `webpackLoader`, `languageTools`.
+
+Community provided tools will be allowed to use their own namespaces (e.g. `elder` or `routify`),
+but these are not supported beyond being tolerated by `@sveltejs/config`
+
+##### validation utilities
+
+SvelteKit contains [code](https://github.com/sveltejs/kit/blob/master/packages/kit/src/core/config/options.js) to validate config options and output helpful messages. 
+Refactor this code to be of general use and provide an api
+
+```js
+import { loadConfig ,validateConfig } from '@sveltejs/config';
+//...
+const config = await load_config();
+const rules = {
+    //TBD, use sveltekit options as a base
+}
+// throws on error
+validateConfig(config, rules);
+```
+
+#### New Features for users
+
+##### `defineConfig` helper for intellisense like in [vite config](https://vitejs.dev/config/#config-intellisense)
+```js
+import { defineConfig } from '@sveltejs/config'
+
+export default defineConfig({
+// ...
+})
+```
+
+##### `async` config support like in [vite config](https://vitejs.dev/config/#async-config)
+```js
+export default defineConfig(async () => {
+    const data = await asyncFunction()
+    return {
+        // build specific config
+    }
+})
+```
+
+##### allow `svelte.config.ts` and transpile automatically
+
+Transpilation is done via lazy dynamic import of `typescript` or `esbuild`. If neither is available, this fails.
+It allows users of typescript projects to fully embrace ts without burdening js projects with a typescript dependency.
+
+
+## How we teach this
+
+### To tool authors
+
+README.md and/or docs/ in the new GitHub repo, example usage in kit and vite-plugin-svelte
+
+### To users
+
+Currently, svelte.config.js isn't really documented on its own. The tools show how they use it.
```

Comment:
maybe just add README.md in `@sveltejs/config` packages folder

Reply:
That would be hard to keep in sync for the different namespaces where the actual type depends on the version used (eg sveltekit3 is going to have config changes that need documenting)

We do have some ideas for improving documentation discoverability though